sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -282,316 +458,356 @@ q.parallel_for(nd_range<2>(G, L), [=](nd_item<2> item)
+|======================
+| Member of `combination` | Description
+|`max_msize`, `max_nsize`, `max_ksize`| if the matrix implementation
+supports a continuous number of element sizes, each of these members
+is non-zero, and the matrix implementation supports all element sizes
+from 1 up to (and including) that number. By contrast, if the matrix
+hardware implementation supports a discrete number of element sizes,
+each of these members has the value zero
+|`msize`, `nsize`, `ksize`| if the matrix implementation supports a
+discrete number of element sizes, each of these members is non-zero,
+and the value tells one of the supported element sizes. By contrast,
+if the matrix hardware supports a continuous number of element sizes,
+each of these members has the value zero
+|`atype`, `btype`, `ctype`, `dtype`| indicates the types supported in
+the combination. these are of type `matrix_type` which tells the list
+of types that are supported for the A, B, and accumulator matrices in
+the `T` template parameter as follows: +
+`bf16`: `sycl::bfloat16` +
+`fp16`: `sycl::half` +
+`tf32`: `sycl::ext::oneapi::experimental::matrix::precision::tf32` +
+`fp32`: `float` +
+`fp64`: `double` +
+`sint8`: `int8_t` +
+`sint16`: `int16_t` +
+`sint32`: `int32_t` +
+`sint64`: `int64_t` +
+`uint8`: `uint8_t` +
+`uint16`: `uint16_t` +
+`uint32`: `uint32_t` +
+`uint64`: `uint64_t`
+|======================
+
+===== Runtime Query Example:
 ```c++
-auto data = get_wi_data(sg, C);
-for (int i = 0; i < data.length; ++i) {
-  auto row, col = data[i].get_coord();
-  sum_of_local_rows[row] += data[i];
+// Ta, Tb, Tc, and Td are the types used in applications
+std::vector<combination> combinations =
+           device.get_info<info::device::matrix::combinations>();
+for (int i = 0; sizeof(combinations); i++) {
+  if (Ta == combinations[i].atype &&
+      Tb == combinations[i].btype &&
+      Tc == combinations[i].ctype &&
+      Td == combinations[i].dtype) {
+    // joint matrix GEMM kernel can be called using these sizes
+    joint_matrix_gemm(combinations[i].msize,
+         combinations[i].nsize, combinations[i].ksize);
+  }
 }
 ```
 
-## TODO List
-- Add WI data to joint matrix mapping coordinates information for piece-wise operations. This will be added as part of the query or new methods to the 'get_wi_data' class. 
-- Add a more realistic and complete example that shows the value of the general query. 
+=== Appendix: Supported Combinations Per Hardware
+
+The table below provides a list of the combinations that
+`joint_matrix` implementations support on each of Intel AMX and Intel
+XMX hardware. Note that these can be returned using
+`ext::oneapi::experimental::info::device::matrix::combinations`.
+
+==== Intel AMX Supported Combinations
+This is currently available in
+`sycl::ext::oneapi::experimental::architecture::intel_cpu_spr`.
+
+[frame="none",options="header"]
+|======================
+| A type | B type | Accumulator type (`ctype` = `dtype`) | M | N | K
```

Comment:
The name of the column "Accumulator type (`ctype` = `dtype`)" seems confusing to me.  I'd suggest the following:

* Change the column name to "C and D type".
* Change the introduction paragraph to say:

  > This is currently available in devices with the architecture `architecture::intel_cpu_spr`.  In this architecture, the C and D matrices support the same set of types, so they are shown in a single column.


Make a similar change below in the table for the section "Intel XMX Supported Combinations".

Reply:
Thinking about this more ... I suppose the type of C and the type of D must be the *same*, right?  If so, the intro should be more like this:

> This is currently available in devices with the architecture `architecture::intel_cpu_spr`. In this architecture, the type of the C matrix must be the same as the type of the D matrix.  Therefore, that common type is shown in a single column in the table below.

(I'm not sure how you want to add the word "implementation", but you can add it in this paragraph too if you want.)